src/hooks/useEndpoint.ts

```diff
@@ -76,11 +79,56 @@ export default function useEndpoint<T>(
             }
         };
 
+        const undoableRequest = () => {
+            let isStopped = false;
+            let sentRequest = false;
+
+            const closeFunction = () => {
+                if (sentRequest === false) {
+                    sentRequest = true;
+                    // eslint-disable-next-line no-void
+                    void request();
+                }
+            };
+
+            window.addEventListener('beforeunload', closeFunction, false);
+            snack(undo?.message || 'Request Sent', {
+                action: React.createElement( 
+                    Button,
+                    {
+                        onClick: () => {
+                            if (isMounted === true) {
+                                _onUndo();
+                                isStopped = true;
+                                setIsLoading(false);
+                                window.removeEventListener('beforeunload', closeFunction, false);
```

Comment:
So from the code we always add the event listener, but then conditionally check if the component is mounted to remove it, correct? I think we should probably remove it regardless of whether or not the component is mounted.  Most of the time I'm checking if the component is mounted so I don't use a `setState` after unmount, but this is a `window.something`, which is global.

Reply:
I moved the removeEventListener call before the isMounted check in onClick. It is also called as the first thing in onExited.